components/tikv_util/src/logger/file_log.rs

```diff
@@ -230,12 +193,117 @@ impl Rotator for RotateBySize {
     }
 }
 
+pub enum Task {
+    Archive,
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "")
+    }
+}
+
+struct LogInfo {
+    f: DirEntry,
+    dt: DateTime<Local>,
+}
+
+pub struct Runner {
+    log_dir: PathBuf,
+    file_name: String,
+    max_backups: usize,
+    max_days: ReadableDuration,
+}
+
+impl Runner {
+    pub fn new<P: AsRef<Path>>(path: P, max_backups: usize, max_days: ReadableDuration) -> Runner {
+        let mut path = path.as_ref().to_owned();
+        let file_name = path.file_stem().unwrap().to_str().unwrap().to_owned();
+        assert!(path.pop());
+        Runner {
+            log_dir: path,
+            file_name,
+            max_backups,
+            max_days,
+        }
+    }
+
+    fn dt_from_file_name(&self, path: &Path) -> Option<DateTime<Local>> {
```

Comment:
Better to add a unit test for this function.

Reply:
And seems this function could be defined as an ordinary function, not a member function

Reply:
Agree, changed to this, and added a unit test. Thanks!